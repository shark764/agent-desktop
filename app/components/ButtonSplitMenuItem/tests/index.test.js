/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { ButtonSplitMenuItem } from '../index';

describe('<ButtonSplitMenuItem />', () => {

  const mockFunc = jest.fn();
  const mockText = {
    id: 'app.containers.ContentAreaTop.wrapupOn',
    defaultMessage: 'Wrap Up On',
  }

  describe('when passed required properties', () => {
    it('should render correctly', () => {
      const rendered = shallow(
        <ButtonSplitMenuItem
          id="mockId"
          clickCallback={mockFunc}
          hideSubMenu={mockFunc}
          text={mockText}
        />
      );
      expect(rendered).toMatchSnapshot();
    });

    describe('when submenu button is clicked', () => {
      it('it should fire the onClick function', () => {
        const rendered = shallow(
          <ButtonSplitMenuItem
            id="mockId"
            clickCallback={mockFunc}
            hideSubMenu={mockFunc}
            text={mockText}
          />
        );

        rendered.find('li').simulate('click');
        expect(mockFunc).toBeCalled();
      });
    });
  });
});
