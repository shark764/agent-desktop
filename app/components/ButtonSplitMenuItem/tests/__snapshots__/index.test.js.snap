// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSplitMenuItem /> when passed required properties should render correctly 1`] = `
<li
  id="button-submenu-listitem-mockId"
  onClick={[Function]}
>
  <FormattedMessage
    defaultMessage="Wrap Up On"
    id="app.containers.ContentAreaTop.wrapupOn"
    values={Object {}}
  />
</li>
`;
